core/src/com/unciv/models/ruleset/tile/TileImprovement.kt

```diff
@@ -68,5 +69,24 @@ class TileImprovement : NamedStats() {
 
     fun hasUnique(unique: String) = uniques.contains(unique)
     fun isGreatImprovement() = hasUnique("Great Improvement")
+
+    /**
+     * Check: Is this improvement allowed on a [given][name] terrain feature?
+     * 
+     * Uses both _legacy_ [resourceTerrainAllow] and unique "Does not need removal of []"
+     * 
+     * Background: This not used for e.g. a lumbermill - it derives the right to be placed on forest
+     * from [terrainsCanBeBuiltOn]. Other improvements may be candidates without fulfilling the
+     * [terrainsCanBeBuiltOn] check - e.g. they are listed by a resource as 'their' improvement.
+     * I such cases, the 'unbuildable' property of the Terrain feature might prevent the improvement,
+     * so this check is done in conjunction - for the user, success means he does not need to remove
+     * a terrain feature, thus the unique name.
+     */
+    fun isAllowedOnFeature(name: String): Boolean {
+        if (name in resourceTerrainAllow) return true
+        return uniqueObjects.filter { it.placeholderText == "Does not need removal of []"
```

Comment:
Instead of .fliter{}.any() you can just .any{}

Reply:
Thanks - shows I didn't yet learn everything about the uniques system...